### PR TITLE
created user posting functionality

### PR DESCRIPTION
**Pull request by madilkhan002**

### CODE REVIEW

File: olx/mobile/urls.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django.urls import path
+from mobile import views
+
+
+app_name = "mobile"
```
Comment (somantalha) [inline olx/mobile/urls.py:5]: Why we need to add `app_name` here. Can we avoid this ?

File: olx/mobile/utils.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+from django.core.paginator import Paginator
+from django.shortcuts import redirect
+
+from .models import Mobile
+
+
+def pagination(request, current_user_posting):
+    DEFAULT_PAGE = 1
+    POSTING_PER_PAGE = 1
```
Comment (somantalha) [inline olx/mobile/utils.py:9]: Can we store these constants in `constants.py`?

File: olx/mobile/views.py [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
 
-# Create your views here.
+from .choices import Condition
+from .utils import (
+    authenticate_user,
+    get_user_posting,
+    perform_filtering,
+    pagination,
+    get_query_params,
+)
```
Comment (somantalha) [inline olx/mobile/views.py:10]: Can we use `absolute imports`.

File: olx/mobile/views.py [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+
+
+def user_posting(request):
+    authenticate_user(request)
```
Comment (somantalha) [inline olx/mobile/views.py:14]: Can we use `@login_required` here ?

File: olx/mobile/views.py [hunk 5]
```diff
@@ -1,3 +1,27 @@
...
+    current_user_posting = get_user_posting(request)
```
Comment (somantalha) [inline olx/mobile/views.py:15]: Can we use query here directly instead of invoking a method just for this?

File: olx/mobile/views.py [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+    filtered_user_posting = perform_filtering(request, current_user_posting)
```
Comment (somantalha) [inline olx/mobile/views.py:16]: Can we use Django filters for this ?

File: olx/mobile/utils.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    paginator = Paginator(current_user_posting, POSTING_PER_PAGE)
+    current_page = request.GET.get('page') or DEFAULT_PAGE
+    postings = paginator.get_page(current_page)
+
+    return postings, paginator, current_page
```
Comment (somantalha) [inline olx/mobile/utils.py:14]: Can we make a dict of these values `postings, paginator, current_page` and return it ?

**Review by somantalha — CHANGES_REQUESTED** [review]

File: olx/mobile/urls.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+urlpatterns = [
+    path("posting/", views.user_posting, name="user_posting"),
```
Comment (UsmanAli99) [inline olx/mobile/urls.py:7]: path("", views.user_mobile_listing, name="mobiles"),

File: olx/mobile/utils.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+from django.core.paginator import Paginator
+from django.shortcuts import redirect
+
+from mobile.models import Mobile
+from mobile.constants import DEFAULT_PAGE, POSTING_PER_PAGE
+
+
+def pagination(request, current_user_posting):
+    paginator = Paginator(current_user_posting, POSTING_PER_PAGE)
+    current_page = request.GET.get('page') or DEFAULT_PAGE
```
Comment (UsmanAli99) [inline olx/mobile/utils.py:10]: Create separate method for this

**Review by UsmanAli99 — CHANGES_REQUESTED** [review]

**Pull request merged**